pilot/pkg/bootstrap/mesh.go

```diff
@@ -16,174 +16,87 @@ package bootstrap
-	log.Infof("version %s", version.Info.String())
-	log.Infof("flags %s", spew.Sdump(args))
-
-	s.mesh = meshConfig
-	return nil
+	// Allow some overrides for testing purposes.
+	if args.Mesh.MixerAddress != "" {
+		meshConfig.MixerCheckServer = args.Mesh.MixerAddress
+		meshConfig.MixerReportServer = args.Mesh.MixerAddress
+	}
+	return mesh.NewFixedWatcher(meshConfig), nil
 }
 
-// initMeshNetworks loads the mesh networks configuration from the file provided
+// newNetworksWatcher loads the mesh networks configuration from the file provided
 // in the args and add a watcher for changes in this file.
-func (s *Server) initMeshNetworks(args *PilotArgs) error { //nolint: unparam
+func newNetworksWatcher(args *PilotArgs, fileWatcher filewatcher.FileWatcher) mesh.NetworksWatcher {
+	var watcher mesh.NetworksWatcher
 	if args.NetworksConfigFile == "" {
 		log.Info("mesh networks configuration not provided")
-		return nil
-	}
-
-	meshNetworks, err := cmd.ReadMeshNetworksConfig(args.NetworksConfigFile)
-	if err != nil {
-		log.Warnf("failed to read mesh networks configuration from %q: %v", args.NetworksConfigFile, err)
-		return nil
-	}
-	log.Infof("mesh networks configuration %s", spew.Sdump(meshNetworks))
-	util.ResolveHostsInNetworksConfig(meshNetworks)
-	log.Infof("mesh networks configuration post-resolution %s", spew.Sdump(meshNetworks))
-	s.meshNetworks = meshNetworks
-
-	// Watch the networks config file for changes and reload if it got modified
-	s.addFileWatcher(args.NetworksConfigFile, func() {
-		// Reload the config file
-		meshNetworks, err := cmd.ReadMeshNetworksConfig(args.NetworksConfigFile)
+	} else {
+		var err error
+		watcher, err = mesh.NewNetworksWatcher(fileWatcher, args.NetworksConfigFile)
 		if err != nil {
-			log.Warnf("failed to read mesh networks configuration from %q", args.NetworksConfigFile)
-			return
+			log.Infoa(err)
 		}
-		if !reflect.DeepEqual(meshNetworks, s.meshNetworks) {
-			log.Infof("mesh networks configuration file updated to: %s", spew.Sdump(meshNetworks))
-			util.ResolveHostsInNetworksConfig(meshNetworks)
-			log.Infof("mesh networks configuration post-resolution %s", spew.Sdump(meshNetworks))
-			s.meshNetworks = meshNetworks
-			if s.kubeRegistry != nil {
-				s.kubeRegistry.InitNetworkLookup(meshNetworks)
-			}
-			if s.multicluster != nil {
-				s.multicluster.ReloadNetworkLookup(meshNetworks)
-			}
-			if s.EnvoyXdsServer != nil {
-				s.EnvoyXdsServer.Env.MeshNetworks = meshNetworks
-				s.EnvoyXdsServer.ConfigUpdate(&model.PushRequest{Full: true})
-			}
-		}
-	})
+	}
 
-	return nil
-}
+	if watcher == nil {
+		log.Info("mesh networks configuration not provided")
+		watcher = mesh.NewFixedNetworksWatcher(nil)
+	}
 
-// Add to the FileWatcher the provided file and execute the provided function
-// on any change event for this file.
-// Using a debouncing mechanism to avoid calling the callback multiple times
-// per event.
-func (s *Server) addFileWatcher(file string, callback func()) {
```

Comment:
where is this being moved to?

Reply:
https://github.com/istio/istio/pull/19167/files#diff-94cf4f94e1d1aac6fe3fd24fbb526bfbR170